### PR TITLE
[aws-json] Migrate SNS GetSubscriptionAttributes

### PR DESCRIPTION
**Pull request by kojisaikiAtSony**

### CODE REVIEW

**Comment by kojisaikiAtSony** [comment]
Hi @Admiral-Piett , this is also ready to review!

File: app/gosns/get_subscription_attributes_test.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		return false
+	}
+
+	_, r := test.GenerateRequestInfo("POST", "/", nil, true)
+	code, _ := GetSubscriptionAttributesV1(r)
+
+	assert.Equal(t, http.StatusBadRequest, code)
+}
+
+func TestGetSubscriptionAttributesV1_succes(t *testing.T) {
```
Comment (Admiral-Piett) [inline app/gosns/get_subscription_attributes_test.go:61]: `_success`?

File: app/gosns/get_subscription_attributes_test.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			SubscriptionArn: subscriptions[1].SubscriptionArn,
+		}
+		return true
+	}
+	_, r := test.GenerateRequestInfo("POST", "/", nil, true)
+	code, response := GetSubscriptionAttributesV1(r)
+
+	result := response.GetResult().(models.GetSubscriptionAttributesResult)
+	assert.Equal(t, http.StatusOK, code)
+	assert.Len(t, result.Attributes.Entries, 9)
```
Comment (Admiral-Piett) [inline app/gosns/get_subscription_attributes_test.go:82]: Might be nice to assert on the values if it's reasonably possible?

File: app/models/errors.go [hunk 3]
```diff
@@ -20,7 +20,7 @@ func init() {
 	SnsErrors = map[string]SnsErrorType{
 		"InvalidParameterValue": {HttpError: http.StatusBadRequest, Type: "InvalidParameterValue", Code: "AWS.SimpleNotificationService.InvalidParameterValue", Message: "An invalid or out-of-range value was supplied for the input parameter."},
 		"TopicNotFound":         {HttpError: http.StatusBadRequest, Type: "Not Found", Code: "AWS.SimpleNotificationService.NonExistentTopic", Message: "The specified topic does not exist for this wsdl version."},
-		"SubscriptionNotFound":  {HttpError: http.StatusBadRequest, Type: "Not Found", Code: "AWS.SimpleNotificationService.NonExistentSubscription", Message: "The specified subscription does not exist for this wsdl version."},
+		"SubscriptionNotFound":  {HttpError: http.StatusNotFound, Type: "Not Found", Code: "AWS.SimpleNotificationService.NonExistentSubscription", Message: "The specified subscription does not exist for this wsdl version."},
```
Comment (Admiral-Piett) [inline app/models/errors.go:23]: Nice!  Thank you!

**Review by Admiral-Piett — COMMENTED** [review]
2 very small things otherwise, looks great!

**Comment by Admiral-Piett** [comment]
@kojisaikiAtSony No rush, but checking in.  Anything else I need to look at or can help with on this?

**Comment by kojisaikiAtSony** [comment]
> @kojisaikiAtSony No rush, but checking in. Anything else I need to look at or can help with on this?

Oh, sorry to leave no reaction. Your feedbacks make sense. We will update this soon 👍

**Comment by kojisaikiAtSony** [comment]
@Admiral-Piett We updated this with your feedback!

**Review by Admiral-Piett — APPROVED** [review]

**Pull request merged**